tools/sim/test/test_carla_integration.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+import os
+import subprocess
+import sys
+import time
+import unittest
+
+import psutil
+
+from common.params import Params
+from selfdrive.manager import manager
+from selfdrive.manager.process import DaemonProcess
+from selfdrive.manager.process_config import managed_processes
+from tools.sim import bridge
+from tools.sim.bridge import connect_carla_client
+
+
+class TestCarlaIntegration(unittest.TestCase):
+  """
+  Tests need Carla simulator to run
+  """
+  subprocess = None
+
+  def test_connect_with_carla(self):
+    # Test connecting to Carla within 5 seconds and return no RuntimeError
+    client = connect_carla_client()
+    assert client is not None
+    # Will raise an error if not connected
+    client.load_world('Town04_Opt')
+
+  def test_run_bridge(self):
+    # Test bridge connect with carla and runs without any errors for 60 seconds
+    test_duration = 60
+
+    params = Params()
+    args = sys.argv[2:]  # Remove test arguments when executing this test
+    params.put_bool("DoShutdown", False)
+
+    p = bridge.main(args, keep_alive=False)[0]
+    time.sleep(test_duration)
+    params.put_bool("DoShutdown", True)
+
+    p.join()
+    # Assert no exceptions
+    self.assertEqual(p.exitcode, 0)
+
+  def assert_processes_running(self, expected_p):
+    running = {p: False for p in expected_p}
+    for p in expected_p:
+      name = managed_processes[p].get_process_state_msg().name
+      for proc in psutil.process_iter():
+        if proc.name().endswith(name):
```

Comment:
@adeebshihadeh This assumes the first argument of the processes in selfdrive/manager/process_config.py
are also used at the end of process name. However just noticing that you are breaking that pattern here: https://github.com/commaai/openpilot/pull/24212
I probably need to change the classes in process_config such that they use the same attribute to define the name of the process.